TINKERPOP-2271 Add warnings=true|false console preference (tp34)

dalaro opened the pull request:
This PR adds a new gremlin-console preference controlling the display of server-originated warnings:

```
:set warnings true // the default; all warnings displayed

:set warnings false // all warnings suppressed
```

https://issues.apache.org/jira/browse/TINKERPOP-2271

spmallette commented:
This approach seems fine to me. Could you please update the reference documentation with this new options:

https://github.com/apache/tinkerpop/blob/1213d3f8276da14ae3854ec16e46b0a2ab2e897e/docs/src/reference/gremlin-applications.asciidoc#console-preferences

A CHANGELOG entry would also make this one fully complete. I don't mind adding both of those items on merge if you can't come back to this one immediately.

VOTE +1

dalaro commented:
Thanks for taking a look and giving guidance.  I just followed up on both of your points, amended the commit, and force-pushed.

I tried to stick to the CHANGELOG file's established convention, so I added an Improvements subsection under 3.4.3 with a single bullet point listing the JIRA key and a one-liner description.

I also updated the asciidoc's table of console prefs.  I couldn't identify a sort on that table, so I just inserted warnings after colors, the only other boolean-typed option.  Here's a link to how it looks on the PR now:

https://github.com/apache/tinkerpop/blob/82de2a60fcff81643008e3b11f5fef358bc8808e/docs/src/reference/gremlin-applications.asciidoc#console-preferences

spmallette commented:
sorry, but for changelog you just needed to add another bullet to the list - the actual "JIRA stuff" gets added on release. thanks

dalaro commented:
I deleted the issue key and the Improvements subsection, so now it's just a bullet with a commit-message-like description.

Also, I just changed the github PR's target branch from master to tp34.  I branched off tp34 when I started working on this.  I just overlooked the target dropdown when opening the PR here on github.

Let me know if you actually wanted this to go into master and didn't want the target changed to tp34.  I think there will be a CHANGELOG conflict.  I can rebase and fix resolve in that case though.

spmallette commented:
hmm - i thought it was already targeted at tp34 when you submitted it, so if you had to change it to be so, then it's in the right place.  Anyway, looks all good now - thank you

robertdale commented:
Might want to add some backticks to the changelog entry.
VOTE +1

dalaro commented:
I just reworded the changelog entry to include the new command in backticks, and I reworded it to adhere more closely to the surrounding style.  I also rebased onto tp34 to fix conflicts with other changelog modifications since my last push.

Is the following consistent with what you had in mind?

```
diff --git a/CHANGELOG.asciidoc b/CHANGELOG.asciidoc
index 24a092b288..bea3b568d0 100644
--- a/CHANGELOG.asciidoc
+++ b/CHANGELOG.asciidoc
@@ -37,7 +37,7 @@ This release also includes changes from <<release-3-3-8, 3.3.8>>.
 * Modified `NumberHelper` to better ignore `Double.NaN` in `min()` and `max()` comparisons.
 * Bump to Netty 4.1.36.
 * Bump to Groovy 2.5.7.
-* Add warnings=true|false console preference
+* Added `:set warnings <true|false>` to Gremlin Console.
 
 [[release-3-4-2]]
 === TinkerPop 3.4.2 (Release Date: May 28, 2019)
```

robertdale commented:
Perfect.

dkuppitz commented:
VOTE +1
